docs/Guidelines.md

```diff
@@ -0,0 +1,77 @@
+# Overview
+The purpose of *this* document is to serve as a primer for developers/qe who are looking for best practices or adding new tests to RHTAP E2E framework.
+
+## General tips
+
+* Make sure you've implemented any required controller functionality that is required for your tests within the following files
+   * `pkg/utils/<new controller directory>` - net new controller logic for a new service or component
+   * `pkg/framework/framework.go` - import the new controller and update the `Framework` struct to be able to initialize the new controller
+* Every test package should be imported to `cmd/e2e_test.go`, e.g. [byoc](https://github.com/redhat-appstudio/e2e-tests/blob/main/cmd/e2e_test.go#L15).
+* Every new test should have correct [labels](docs/LabelsNaming.md).
+* Every test should have meaningful description with JIRA/GitHub issue key.
```

Comment:
This is not clear to me. Say, we added a new functionality to our operator and have to test it. What should be added in such case? GitHub PR link?

Reply:
That is up to your team. You can keep the link to the initial feature request or update with the new JiraKey